### DISCUSSION TITLE
Android auto with mobile tethered for internet

### DISCUSSION THREAD

**Discussion by NKPenfold** [tesla-android/issue-tracker#314] [Q&A] [answered]
I've installed the latest version of Tesla Android and it is working really well with my mobile tethered by usb for internet access. Does anyone know if I can add a carlinkit ccpa to give me android auto access while my phone is also tethered for internet? If so would this then enable me to use voice commands like for replying to texts/WhatsApp and asking maps to navigate to a location

**Comment by brswattt** [comment]
I am going to test this tomorrow and I'll let you know.

Did you just download maps and other apps on TeslaAndroid and not using Android Auto via other means such as the Headunit reloaded / TesAA app?

**Reply by NKPenfold** [reply]
Excellent, look forward to hearing how you get on. 

To answer your question; yes, I've installed Google maps and Waze on Tesla android and am using them directly. No headunit reloaded or TeslaAA.

I have found a bit of a workaround. I bought a cheap (£7) wireless usb microphone and plugged its dongle in to the pi. That now lets me talk to the maps apps on Tesla android and use Google assistant whilst my phone is still usb tethered for internet access

**Reply by brswattt** [reply, marked as answer]
It works just fine, Tesla voice commands and all with USB tethering / carlinkit dongle. What I don't like is the resolution of the dongle, even though its set to 1080p. everything just looks fuzzy.

**Reply by brswattt** [reply]
Changed the developer settings of Android auto on my phone and set the carlink autokit settings to "Original" / "L" Screen Size / 1080p / 60 fps and it seems more clear.

**Reply by brswattt** [reply]
Can't get the Google assistant to pick up my voice though. Tried both carlink mic settings.

**Reply by brswattt** [reply]
Honestly out of my hours of testing and fiddling with settings, I've come to the conclusion that what is the fastest/works the best for me at the moment is the TesAA app. I can't get the carlinkkit to look as good as the app. Native android on my Pi4b is just slow compared to the AA auto experience TesAA provides. Of course, I'll continue to see what works the best but if I can get the dongle to look as good as TesAA, I'd prob use that. It was fast enough.

**Reply by NKPenfold** [reply]
That's fantastic, thanks for your help and feedback

**Reply by brswattt** [reply]
I also tried the "Headunit reloaded" app and experienced the same "not quite as crisp" android auto experience, but with that app I could not get the USB tether to work at the same time as the wired android auto. I'm starting to wonder if the bluryness is with TeslaAndroid, even though I have the resolution set to 832p, that's just how it's going to look. In the settings menu it looks nice and sharp, but when you get into the android OS....its just blurry. I wonder if it's a pixel density/DPI thing? TesAA has a 1080p setting so I wonder if its just downscaling in the browser, that's why it looks better....not sure how it works under the hood. With the wireless through the HUR app, I could not get mobile data to pass through the wifi hotspot connection provided by TeslaAndroid, as whenever I plugged the cable in to try to tether, it would auto connect via wired. I tried to block it in the settings from connecting to my phone but I could not figure it out. Even with the circle with the line through it next to my phone name under the 'USB' setting. No Native AA experience kind of sucks, and just browsing around the native google maps app through TeslaAndroid was just slow compared to TesAA. TesAA also has browser audio, but I haven't fiddled around with that much yet. One thing I did notice about TeslaAndroid, is that it seemed like my music was louder. Maybe it was just me.....so take that with a grain of salt. 2023 M3 RWD. Pixel 6a.

**Reply by brswattt** [reply]
Wanted to give another update, on the new method that works for me. I have changed my mind on HeadUnit reloaded.

Context: 2023 Tesla Model 3. Pixel 6a with Visible+.
I was using TesAA for a bit, but when it decided it was time to download a software update for Tesla and was disconnecting all the time, I decided against it. When it was NOT downloading an update in the background, it worked good. Browser audio was a little too bandwidth intensive in my opinion so I opted to use BT audio. Note: TeslaAndroid has the options to block telemetry/software updates and TesAA doesn't. Conclusion: Great app in 2023 Model 3 / Pixel 6a when not downloading something in the background on the Tesla.

That wasn't going to cut it for me, so I started looking into headunit reloaded again. I put my Pi in a plastic case with a fan/heatsink (Argon Poly+). I first tested TeslaAndroid in offline mode with headunit reloaded. I must have been mistaken on the bluriness between TesAA and headunit reloaded, it seems the same to me. The CCPA dongle was the only one I could not get looking good, despite being set to 1080p.

Went ahead and plugged my phone in via USB to the pi, and headunit went through the android auto setup and displayed it on the screen. Only issue I had (Which I still don't know why it happens), is when I unplugged my phone from the Pi to stop Android Auto, it would disable my TeslaAndroid hotspot! Not sure why. I had to go into the app settings and disable "Modify system settings" permission. It will give you a warning, but I haven't had any issues after disabling that. Another issue I ran into was that I would plug my phone in, but AA wouldn't start up in headunit reloaded. Not sure what fixed it, but I enabled USB debugging on my phone and in the AA developer options on my phone, I set Android Auto to "Always" start. I unplugged and plugged my phone in multiple times back to back and have not had a hotspot disable or had to manually select the "USB". I'm sure it will bug out at some point making me hit the device in the usb menu in HUR again, but fingers crossed this fixes it. Edited audio/gps/graphics settings in HUR to my liking. Looks good! Movement from the browser is smooth and pretty responsive. Now, I have to find a solution to get internet to my TeslaAndroid because I want to use Tesla Voice commands. I refuse to buy a new sim, or a new data plan. So I bought a gl-inet travel router (Mango). this thing is like $20 and is so awesome. Runs on openWRT. VPN and all of that. Micro USB powered, small. I have Automate on my Pixel 6a, and have a flow set that toggles my WiFi hotspot when BT connects to my car. This travel router acts in "repeater" mode and connects to said WiFi network. It has 2 ethernet ports on it, one for WAN and LAN. I plug a 0.5ft ethernet cable from the LAN port to my Pi, and it provides internet no problem for voice commands, etc. Now, my last problem is that the charging speed from the Pi to my phone is slow as hell. I have seen solutions to this using a USB-C PD splitter, and have one ordered and will let you know my experiences on that.

**Reply by NKPenfold** [reply]
Fantastic work! Thanks for the update
